File: python/readability.py
# Gets user input
text = input("Text: ").lower()

# Splits text into words array
words = text.split()

# Count letters
letters = 0
for word in words:
    for letter in word:
        if letter >= "a" and letter <= "z":
            letters += 1

# Average letters per 100 words
L = letters / (len(words) / 100)

# Count sentences
sntc = 0
for char in text:
    if char == "." or char == "?" or char == "!":
        sntc += 1

# Average sentences per 100 words
S = sntc / (len(words) / 100)

# Calculate grade
grade = (0.0588 * L) - (0.296 * S) - 15.8

if (grade >= 16):
    print("Grade 16+\n")
elif (grade < 1):
    print("Before Grade 1\n")
else:
    print(f"Grade {round(grade)}\n")

File: cs50x/C (lang)/readability.c
#include <cs50.h>
#include <string.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Asks the user to input text and returns it to str variable
    string str = get_string("Text: ");

    // Variables for counts
    int letters = count_letters(str);
    int words = count_words(str);
    int sentences = count_sentences(str);

    // Calculates average letters/sentences per 100 words
    double L = letters / (words / 100.0);
    double S = sentences / (words / 100.0);

    // Calculates Coleman-Liau index and rounds it to int
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    // If index less than 16 and more than 1 prints index
    if (index < 16 && index > 1)
    {
        printf("Grade %d\n", index);
    }

    // If index less than 1 prints "Before Grade 1"
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    // If index more than 16 prints "Grade 16+"
    else
    {
        printf("Grade 16+\n");
    }
    return 0;
}

// Counts letters in given string
int count_letters(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (tolower(text[i]) >= 'a' && tolower(text[i]) <= 'z')
        {
            count++;
        }
    }
    return count;
}

// Counts words in given string
int count_words(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ' && text[i + 1] != ' ')
        {
            count++;
        }
    }
    return count + 1;
}

// Counts sentences in given string
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count++;
        }
    }
    return count;
}
